### PR TITLE
[WIP] Create Button in relation DropDown

### PR DESCRIPTION
**Pull request by marcellovetro**
My PR is a:
💅 Enhancement 

Main update on the:
Plugin: Content Manager

This is a first Demo to Add a Create Button to SelectMany to add a related Content Type record .
This is linked to the issue #1965

### CODE REVIEW

File: packages/strapi-plugin-content-manager/admin/src/containers/EditPage/index.js [hunk 1]
```diff
@@ -334,6 +334,36 @@ export class EditPage extends React.Component {
...
+      type: 'select'
+    };
+
+    this.props.match.params = {
+      slug: relation.collection,
+      id: 'create'
+    };
+
+    this.props.resetProps();
+    this.componentDidMount();
```
Comment (soupette) [inline packages/strapi-plugin-content-manager/admin/src/containers/EditPage/index.js]: I know that this is a WIP but we have recoded the `content-manager` plugin in order to prepare its extensibility (allowing to add custom code without modifying the core one).
Therefore, we start to follow  conventions and tried to use as much functional programming as possible.

As the one responsible for the front-end I can't allow `this.componentDidMount()` in the codebase, you should either use the `componentDidUpdate()` lifecycle or the sagas.

File: packages/strapi-plugin-content-manager/admin/src/containers/EditPage/index.js [hunk 2]
```diff
@@ -334,6 +334,36 @@ export class EditPage extends React.Component {
...
+      value: (relation.nature == 'oneToMany' ? this.props.editPage.record: [this.props.editPage.record]),
+      type: 'select'
+    };
+
+    this.props.match.params = {
+      slug: relation.collection,
+      id: 'create'
+    };
+
+    this.props.resetProps();
```
Comment (soupette) [inline packages/strapi-plugin-content-manager/admin/src/containers/EditPage/index.js]: I'm sure that there's another way to do that, you should use a lifecycle

File: packages/strapi-plugin-content-manager/admin/src/components/SelectMany/index.js [hunk 3]
```diff
@@ -159,16 +168,35 @@ class SelectMany extends React.PureComponent {
     );
     const value = get(this.props.record, this.props.relation.alias) || [];
 
+    //Create Button create template and check configuration
+    const attrSchema =  this.props.schema.attributes[this.props.relation.alias];
+    console.log('this.props',this.props);
+    console.log('attrSchema',attrSchema);
```
Comment (copeus) [inline packages/strapi-plugin-content-manager/admin/src/components/SelectMany/index.js]: remove debug...

**Comment by marcellovetro** [comment]
Ok I've just developed the feature in the branch "marcellovetro:create_button"  but it need to be integrated with the content manager interface editor, i the next days I will merge the branch

**Comment by soupette** [comment]
Hi @marcellovetro what's the status of your PR?

**Comment by marcellovetro** [comment]
I've completed a part of work in the create_button branch and it works great!!  I managed these options: 
1. createRelated: show the link "Add" and precompile the relation, when the insert was done, it jump to the origin
2.  hideSelect: to hide the Drow Down than add an existing element. Example: if I manage a Bill I wont only create a detail but I won't select an existing one

But it need move the configuration from the <model>.settings.json into the model to the content manager plugin

**Comment by lauriejim** [comment]
Since there is no activity on this PR, I close it.
We will reopen it if it's needed.

**Pull request closed without merging**